### PR TITLE
Allow instructors to enable / disable the offline IDE usage in programming exercises

### LINKED ISSUE #1661: Add option for programming exercises to enable/disable offline mode
Currently, instructors can enable / disable the use of the online code editor in programming exercises.

However, they cannot enable / disable the offline mode based on git and the IDE (where students clone the repository).

For the online exam, it could make senes to disable the offline mode. Then students won't see the "clone repository" button any more. In addition, they would not get access to their own repository, when they start the exercise.

This customization option should be a new checkbox in the programming exercise creation / editing dialog.

Instructors can enable both options or only one of them (offline vs. online), however they cannot disable both options, because then no student can participate in the programming exercise.

**Simplification:** When instructors change the offline mode during edit, this will not affect already created participations (changing the access control for all existing student repositories can be implemented in a second step).

### PR DESCRIPTION
**Pull request by krusche**
Fixes #1661 

### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Client: I translated all the newly inserted strings into German and English

### Motivation and Context
Instructors might want to create programming exercises where students can only use the online code editor and disallow the use of the offline IDE (and git). This was not possible until now.

### Description
I added a new option to allow the offline IDE usage (by default this is true).
In case this option is set to false by the instructor, the students will not get access to the git repository when they start the exercise and they won't see the clone repository button.

### Steps for Testing
1. Create a programming exercise with "Allow offline IDE" disabled
2. Start the programming exercise
3. Double check that the clone repository button is never visible
4. (You should also not be able to import this exercise into Orion)
5. Double check that you do not have access rights to the git repository
6. Change the configuration to "Allow offline IDE" enabled
7. Delete your participation for this programming exercise (including its repository)
8. Start the exercise again
9. Now the clone button should be visible, you should have access rights in Bitbucket and you should be able to clone the repository

Note: there are some limitations when instructors change the value after some participations have been create (i.e. students started to work on the programming exercise). Then the access for existing repositories is not changed. This is out of scope for this PR and should be improved in a future PR.

### CODE REVIEW

File: src/main/java/de/tum/in/www1/artemis/service/connectors/VersionControlService.java [hunk 1]
```diff
@@ -12,7 +12,13 @@
 
 public interface VersionControlService {
 
-    void configureRepository(URL repositoryUrl, Set<User> users);
+    /**
+     * Configure the repository for the student(s), this mainly includes
```
Comment (sleiss) [inline src/main/java/de/tum/in/www1/artemis/service/connectors/VersionControlService.java:16]: I believe some parts of the documentation got lost here

File: src/main/webapp/app/overview/exercise-details/programming-exercise-student-ide-actions.component.ts [hunk 2]
```diff
@@ -32,7 +32,7 @@ export class ProgrammingExerciseStudentIdeActionsComponent implements OnInit {
     @Input() @HostBinding('class.col') equalColumns = true;
     @Input() @HostBinding('class.col-auto') smallColumns = false;
 
-    @Input() exercise: Exercise;
+    @Input() exercise: ProgrammingExercise;
```
Comment (sleiss) [inline src/main/webapp/app/overview/exercise-details/programming-exercise-student-ide-actions.component.ts:35]: I believe the import above can now be removed (as github already detected), I can't add the review in the import statement for some reason (maybe one can only add it to changed lines).

File: src/main/java/de/tum/in/www1/artemis/service/connectors/VersionControlService.java (hunk 1 above)
Comment (krusche) [reply src/main/java/de/tum/in/www1/artemis/service/connectors/VersionControlService.java:16]: thanks, I fixed it

File: src/main/webapp/app/overview/exercise-details/programming-exercise-student-ide-actions.component.ts (hunk 2 above)
Comment (krusche) [reply src/main/webapp/app/overview/exercise-details/programming-exercise-student-ide-actions.component.ts:35]: removed the import

**Pull request merged**